style(SignIn&SignUp): add styles for sign in and signup pages

HaiderAltufaily opened the pull request:
## What?
Created Sign In and Sign Up pages along with their components



## Note:
In order to view the changes go to `/signin` or `/signup` in the preview page

HaiderAltufaily committed:
style(Signin&SignUp): add styles for sign in and signup pages

EsraaY21 commented:
Looks amazing! Great work =)

MuhannedNoman commented on components/SignIn/ForgotPass.js:
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable react/no-children-prop */
```
Why are we disabling eslint here?

HaiderAltufaily replied:
In chakra ui you need to use children as a prop and eslint considers this as an error so I had to disable it 
ex:
`  <InputLeftElement
                    pointerEvents="none"
                    fontSize="lg"
                    h="46px"
                    children={<BiEnvelope size="20" />}
                  />`
This gave me an error because I used children as a prop here....
So I only disabled the rule for children props

HaiderAltufaily replied:
@MuhannedNoman Should I look for another solution to this problem?

MuhannedNoman replied:
Can you disable it for next line only.

HaiderAltufaily replied:
Okay I will do that

MuhannedNoman reviewed (approved):
Hey @HaiderAltufaily nice work here, but why is eslint disabled in a few components?

HaiderAltufaily committed:
solve theme conflict

HaiderAltufaily committed:
Merge branch 'dev' of https://github.com/Iraq-RBC-Capstones-2021/capstone-react-lets-work into HaiderAltufaily/issue7

MuhannedNoman commented:
Also, @HaiderAltufaily what are you using for validating the data like password!?

HaiderAltufaily committed:
fix(eslint):disable eslint for few lines only instead of full components

HaiderAltufaily commented:
Umm, I thought I should only make the design and I didn't do anything for the validation yet... I was planning to do them when I start working on the firebase authentication as firebase has its own email&pass validation and error messages...
So should I work on validation now? @MuhannedNoman

HaiderAltufaily committed:
feat(hooks): add custom input hook (useInput) to handle form validation

HaiderAltufaily committed:
feat(chakraUI): add custom primary and error input variants

HaiderAltufaily committed:
feat(SignIn&SignUp): add form validation

HaiderAltufaily commented:
I have created a custom input hook and validated each data with error messages... and also I have made custom Input styling (both primary and error variants). More error messages will be added once we work on the firebase authentication...
With these changes, can I have a new review..?

EsraaY21 reviewed (changes requested):
Great work! The sing up page works perfectly but there is a small problem with the password validation in the sign in page. It doesn't show me an error when I type a password smaller than 6.

HaiderAltufaily committed:
fix(SignIn):fix password validation

EsraaY21 reviewed (approved):
Nice! LGTM =)

HaiderAltufaily committed:
perf(forgotPass): remove forgotPass.js now, only the forms change
instead of having a separate forgotpass.js when clicking on forgot pass
it will conditionally display the forgotpass form instead of the sign in form

HaiderAltufaily committed:
fix(conflicts)

MuhannedNoman commented:
> 
> 
> I have created a custom input hook and validated each data with error messages... and also I have made custom Input styling (both primary and error variants). More error messages will be added once we work on the firebase authentication... With these changes, can I have a new review..?

This is only a simple validation, using a library you can do a more complicated validation for symbol, number of letters, ...etc. 
Also, using a library here is considered front end validation so it has nothing to do with firebase or any other service used for the back-end

HaiderAltufaily committed:
perf(SignIn&SignUp Translation): add arabic translation
add arabic translation with the appropriate styling

HaiderAltufaily committed:
feat(formik): add formik and yup validation

HaiderAltufaily commented:
Hey @MuhannedNoman I have refactored the code and these are the main changes:
## Changes:
- Deleted the custom hook and instead used formik and yup for validation
- created a shared folder inside the components, inside this folder I have created a `ChakraInput` and `ChakraCheckbox` in these components I have utilized chakra ui `Input` with the `Field` component of formik.
- Used `ChakraInput` inside sign in and sign up pages which reduced the code a lot
- Recreated `forgotPass.js` but it's simpler now.
- added arabic language for both of the pages (Didn't add it for the error messages yet) to check go to: 
https://capstone-react-lets-work-git-haideraltufaily-issue7-lets-work.vercel.app/ar/signin

MuhannedNoman reviewed (approved):
LGTM nice work here @HaiderAltufaily

HaiderAltufaily committed:
fix(conflicts)

EsraaY21 reviewed (approved):
Great Work!

HaiderAltufaily committed:
fix(notifications): fix notifications position
